fairseq/utils.py

```diff
@@ -452,7 +453,7 @@ def import_user_module(args):
         # a deterministic hash of the original module_path.
         with tempfile.TemporaryDirectory() as tmpdirname:
             unique_mod_name = "fairseq_user_dir_{}".format(hash(module_path) % 100000)
-            os.symlink(module_path, os.path.join(tmpdirname, unique_mod_name))
+            shutil.copytree(module_path, os.path.join(tmpdirname, unique_mod_name))
```

Comment:
Ah, this is probably not what's intended -- in the case of a user passing in `--user-dir` we don't want to copy the whole thing to a temporary directory. The goal here is just to import it under a unique name, and unfortunately importlib.import_module doesn't seem to support this, so we hacked it with symlinks.

But I thought symlinks created in this way would work in Windows, is that not the case?

Reply:
Symlinks require admin privileges in Windows, so this won't work as-is. One option might be to try to symlink and fall back to copy in the Windows case. Another is to simply make this unsupported on Windows.

Reply:
Also, this is tree copy is functionally equivalent since the directory gets cleaned up. I guess it _could_ behave differently if your `--user-dir` is, say, really really big since the copy might take some time. 😄  Are there other concerns?

Reply:
Ah, I see. How about a try/except and fallback to copytree if the symlink fails

Reply:
That seems reasonable to me!